src/main/java/com/springbootassignment/cruddemo/service/CourseService.java

```diff
@@ -0,0 +1,22 @@
+package com.springbootassignment.cruddemo.service;
+
+
+import com.springbootassignment.cruddemo.entity.Course;
+
+
+import java.util.List;
+
+public interface CourseService {
+
+    public List<Course> findAll();
+
+    public Course findById(int theId);
+
+    public void save(Course theCourse);
```

Comment:
Use DTO here insted of entities as an argument.
Update in all other places as well , send dto as an argument from the controller and in the service method map that dto to respective entity and perform ops in db